### DISCUSSION TITLE
Function with label

### DISCUSSION THREAD

**Discussion by erwindenboer** [ManimCommunity/manim#1793] [Q&A]
Hello,

I want to put a graph with a label on the screen. My code is as follows:

`    from manim import *
     from manim_presentation import Slide

     class Begin(Slide):
       def construct(self):
           ax = Axes(
               x_range=[-2, 5, 1],
               y_range=[-5, 5, 1],
               x_length=12,
               y_length=7,
               x_axis_config={"numbers_to_include": np.arange(-2, 6, 1)},
               y_axis_config={"numbers_to_include": np.arange(-5, 6, 1)},
               tips=False
           )

     labels = ax.get_axis_labels()
     f1 = lambda x: x**3-4*x**2+3*x+1
     func_1 = FunctionGraph(f1, color=RED)      
     func_label = ax.get_graph_label(func_1, label=func_1, direction= 2*RIGHT, x_val=1)

     self.play(FadeIn(ax, func_1, func_label))
     self.pause()
     self.wait()
`

I put my function outside the FunctionGraph(), so I can use it for the label. However, when i compile this code it gives the following error:

`ValueError: Alpha -2 not between 0 and 1`

I have no idea what this means. Can someone help me debug this?

**Comment by hydrobeam** [comment]
Could you post the full traceback?

**Reply by erwindenboer** [reply]
@hydrobeam Thanks for the fast response. Here is the full traceback:

PS C:\Users\Erwin\Desktop\project> manim render -pqh --fps 144.0 c:\Users\Erwin\Desktop\project\test.py
Manim Community v0.8.0

╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ c:\users\erwin\appdata\local\programs\python\python39\lib\site-packages\manim\cli\render\command │
│ s.py:135 in render                                                                               │
│                                                                                                  │
│   132 │   │   for SceneClass in scene_classes_from_file(file):                                   │
│   133 │   │   │   try:                                                                           │
│   134 │   │   │   │   scene = SceneClass()                                                       │
│ ❱ 135 │   │   │   │   scene.render()                                                             │
│   136 │   │   │   except Exception:                                                              │
│   137 │   │   │   │   error_console.print_exception()                                            │
│   138 │   │   │   │   sys.exit(1)                                                                │
│                                                                                                  │
│ c:\users\erwin\appdata\local\programs\python\python39\lib\site-packages\manim_presentation\slide │
│ .py:52 in render                                                                                 │
│                                                                                                  │
│   49 │   │   max_files_cached = config["max_files_cached"]                                       │
│   50 │   │   config["max_files_cached"] = float("inf")                                           │
│   51 │   │                                                                                       │
│ ❱ 52 │   │   super(Slide, self).render(*args, **kwargs)                                          │
│   53 │   │                                                                                       │
│   54 │   │   config["max_files_cached"] = max_files_cached                                       │
│   55                                                                                             │
│                                                                                                  │
│ c:\users\erwin\appdata\local\programs\python\python39\lib\site-packages\manim\scene\scene.py:210 │
│ in render                                                                                        │
│                                                                                                  │
│    207 │   │   """                                                                               │
│    208 │   │   self.setup()                                                                      │
│    209 │   │   try:                                                                              │
│ ❱  210 │   │   │   self.construct()                                                              │
│    211 │   │   except EndSceneEarlyException:                                                    │
│    212 │   │   │   pass                                                                          │
│    213 │   │   except RerunSceneException as e:                                                  │
│                                                                                                  │
│ c:\Users\Erwin\Desktop\project\test.py:50 in construct                                           │
│                                                                                                  │
│   47 │   │   #horz_line_1.height = func_1(2.2153)                                                │
│   48 │   │   #horz_line_1.init_colors = GREEN                                                    │
│   49 │   │                                                                                       │
│ ❱ 50 │   │   ax.get_graph_label(func_1, label=func_1, direction= 2*RIGHT, x_val=1)               │
│   51 │   │   #func_label_1 = ax.get_graph_label(func_2, label="3x^2-8x+3", direction= 2*LEFT,    │
│   52 │   │                                                                                       │
│   53 │   │   self.play(FadeIn(ax, func_1))                                                       │
│                                                                                                  │
│ c:\users\erwin\appdata\local\programs\python\python39\lib\site-packages\manim\mobject\coordinate │
│ _systems.py:533 in get_graph_label                                                               │
│                                                                                                  │
│    530 │   │   │   │   if point[1] < config["frame_y_radius"]:                                   │
│    531 │   │   │   │   │   break                                                                 │
│    532 │   │   else:                                                                             │
│ ❱  533 │   │   │   point = self.input_to_graph_point(x_val, graph)                               │
│    534 │   │                                                                                     │
│    535 │   │   label.next_to(point, direction, buff=buff)                                        │
│    536 │   │   label.shift_onto_screen()                                                         │
│                                                                                                  │
│ c:\users\erwin\appdata\local\programs\python\python39\lib\site-packages\manim\mobject\coordinate │
│ _systems.py:462 in input_to_graph_point                                                          │
│                                                                                                  │
│    459 │   │   if hasattr(graph, "underlying_function"):                                         │
│    460 │   │   │   return graph.function(x)                                                      │
│    461 │   │   else:                                                                             │
│ ❱  462 │   │   │   alpha = binary_search(                                                        │
│    463 │   │   │   │   function=lambda a: self.point_to_coords(graph.point_from_proportion(a))[  │
│    464 │   │   │   │   │   0                                                                     │
│    465 │   │   │   │   ],                                                                        │
│                                                                                                  │
│ c:\users\erwin\appdata\local\programs\python\python39\lib\site-packages\manim\utils\simple_funct │
│ ions.py:96 in binary_search                                                                      │
│                                                                                                  │
│    93 │   rh = upper_bound                                                                       │
│    94 │   while abs(rh - lh) > tolerance:                                                        │
│    95 │   │   mh = np.mean([lh, rh])                                                             │
│ ❱  96 │   │   lx, mx, rx = [function(h) for h in (lh, mh, rh)]                                   │
│    97 │   │   if lx == target:                                                                   │
│    98 │   │   │   return lx                                                                      │
│    99 │   │   if rx == target:                                                                   │
│                                                                                                  │
│ c:\users\erwin\appdata\local\programs\python\python39\lib\site-packages\manim\utils\simple_funct │
│ ions.py:96 in <listcomp>                                                                         │
│                                                                                                  │
│    93 │   rh = upper_bound                                                                       │
│    94 │   while abs(rh - lh) > tolerance:                                                        │
│    95 │   │   mh = np.mean([lh, rh])                                                             │
│ ❱  96 │   │   lx, mx, rx = [function(h) for h in (lh, mh, rh)]                                   │
│    97 │   │   if lx == target:                                                                   │
│    98 │   │   │   return lx                                                                      │
│    99 │   │   if rx == target:                                                                   │
│                                                                                                  │
│ c:\users\erwin\appdata\local\programs\python\python39\lib\site-packages\manim\mobject\coordinate │
│ _systems.py:463 in <lambda>                                                                      │
│                                                                                                  │
│    460 │   │   │   return graph.function(x)                                                      │
│    461 │   │   else:                                                                             │
│    462 │   │   │   alpha = binary_search(                                                        │
│ ❱  463 │   │   │   │   function=lambda a: self.point_to_coords(graph.point_from_proportion(a))[  │
│    464 │   │   │   │   │   0                                                                     │
│    465 │   │   │   │   ],                                                                        │
│    466 │   │   │   │   target=x,                                                                 │
│                                                                                                  │
│ c:\users\erwin\appdata\local\programs\python\python39\lib\site-packages\manim\mobject\types\vect │
│ orized_mobject.py:1085 in point_from_proportion                                                  │
│                                                                                                  │
│   1082 │   │   """                                                                               │
│   1083 │   │                                                                                     │
│   1084 │   │   if alpha < 0 or alpha > 1:                                                        │
│ ❱ 1085 │   │   │   raise ValueError(f"Alpha {alpha} not between 0 and 1.")                       │
│   1086 │   │                                                                                     │
│   1087 │   │   self.throw_error_if_no_points()                                                   │
│   1088 │   │   if alpha == 1:                                                                    │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
ValueError: Alpha -2 not between 0 and 1.
